xbmc/platform/darwin/osx/XBMCApplication.mm

```diff
@@ -268,7 +231,7 @@ - (BOOL)application:(NSApplication*)theApplication openFile:(NSString*)filename
   if (arg == nullptr)
     return FALSE;
 
-  newargv = (char**)realloc(gArgv, sizeof(char*) * (gArgc + 2));
+  newargv = (const char**)realloc(gArgv, sizeof(char*) * (gArgc + 2));
```

Comment:
Possible to avoid c style casts and use c++ casts instead? Seems to be possible according to other objective c files we have

Reply:
I agree that blind casts (c-style) should be avoided, but I was only addressing the const virality and didn't want to add distracting changes.

I have another topic branch which replaces this code with _NSGetArgv() so I'd be inclined not to make extra changes here.

Reply:
Well it's not mandatory, just a suggestion. We usually take those opportunities to partially modernize the codebase (rename stuff according to code guidelines, update casts, remove NULL in favor of nullptr, etc). But it's perfectly fine if you don't want to bother with that stuff.